wordpress-mobile/WordPress-Android
# [Reader Improvements] Implement new design for tags section

Fixes #19083 

Design specs: zQnohyMpLzBzQ5jzMMKni3-fi

To test:
1 - Run JP and sign in;
2 - Open debug settings and enable `ReaderImprovementsFeatureConfig` FF;
3 - Open reader;
4 - Select "Discover" tab;
5 - Scroll until you find the "You might like" tags section;
6 - Verify the layout for light and dark themes: the tags section should have the new design. Make sure that clicking on tags action still works as expected;
7 - Repeat step 2 but this time disable the FF;
8 - Repeat steps 3-5 and verify the layout: the section layout should appear like it currently does in prod;

## Regression Notes
1. Potential unintended areas of impact
Existing tags section on Reader

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Updated `ReaderDiscoverViewModelTest.kt` but changes were mainly implemented on views.

PR submission checklist:


UI Changes testing checklist:
